approvals/src/main/java/no/sikt/nva/approvals/domain/ApprovalServiceImpl.java

```diff
@@ -90,8 +91,31 @@ public Approval getApprovalByNamedIdentifier(NamedIdentifier namedIdentifier)
     @JacocoGenerated
     @Override
     public Approval updateApprovalIdentifiers(UUID approvalId, Collection<NamedIdentifier> namedIdentifiers)
-        throws ApprovalNotFoundException, ApprovalServiceException, ApprovalConflictException {
-        return null;
+        throws ApprovalNotFoundException, ApprovalServiceException, ApprovalConflictException, RepositoryException {
+        var identifiers = approvalRepository.findIdentifiers(namedIdentifiers);
+        var approval = getApprovalByIdentifier(approvalId);
+
+        ensureIdentifiersAreNotUsedByOtherApproval(identifiers, approval);
+        var updatedApproval = new Approval(approval.identifier(), namedIdentifiers, approval.source(), approval.handle());
+        approvalRepository.updateApprovalIdentifiers(updatedApproval);
```

Comment:
As discussed, this implementation is vulnerable to race conditions that could lead to data loss and/or data corruption. There are multiple problems here:

1. No safeguard against concurrent updates/inserts between `ensureIdentifiersAreNotUsedByOtherApproval(identifiers, approval)` and the actual database write operation.
2. New identifiers added in `no.sikt.nva.approvals.persistence.DynamoDbApprovalRepository#updateApprovalIdentifiers` are written with a simple `PUT` with no conditions, unlike the implementation of the very similar case in `no.sikt.nva.approvals.persistence.DynamoDbApprovalRepository#save`.